tests/integration/kubernetes/k8s-guest-pull-image-signature.bats

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env bats
+# Copyright (c) 2024 IBM Corporation
+# Copyright (c) 2024 Alibaba Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_kbs.sh"
+
+export KBS="${KBS:-false}"
+
+setup() {
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
+
+    setup_common
+    UNSIGNED_UNPROTECTED_REGISTRY_IMAGE="quay.io/prometheus/busybox:latest"
+    UNSIGNED_PROTECTED_REGISTRY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:unsigned"
+    COSIGN_SIGNED_PROTECTED_REGISTRY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:cosign-signed"
+    COSIGNED_SIGNED_PROTECTED_REGISTRY_WRONG_KEY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:cosign-signed-key2"
+}
+
+function setup_kbs_image_policy() {
+    if [ "${KBS}" = "false" ]; then
+        skip "Test skipped as KBS not setup"
+    fi
+
+    default_policy="${1:-insecureAcceptAnything}"
+
+    policy_json=$(echo "{
+    \"default\": [
+        {
+            \"type\": \"${default_policy}\"
+        }
+    ],
+    \"transports\": {
+        \"docker\": {
+            \"ghcr.io/confidential-containers/test-container-image-rs\": [
+                {
+                    \"type\": \"sigstoreSigned\",
+                    \"keyPath\": \"kbs:///default/cosign-public-key/test\"
+                }
+            ],
+            \"quay.io/prometheus\": [
+                {
+                    \"type\": \"insecureAcceptAnything\"
+                }
+            ]
+        }
+    }
+}")
+
+    public_key=$(echo "-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEwQEjdCiL3ILUf07NDkDVhgKCj1C6
+BsCfmM/zt1kNSj0/+nAqA+25XfyClYq2lJFJ6TkgCsf57cTCkXYDz9c+Yg==
+-----END PUBLIC KEY-----")
+
```

Comment:
Similar cosmetic improvement :
```
public_key=$(cat<<EOF
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEwQEjdCiL3ILUf07NDkDVhgKCj1C6
BsCfmM/zt1kNSj0/+nAqA+25XfyClYq2lJFJ6TkgCsf57cTCkXYDz9c+Yg==
-----END PUBLIC KEY-----
EOF
)
```

but most importantly, where does this public key come from ? How can we ensure it will _work_ ?

Reply:
yes, this pubkey comes from https://github.com/confidential-containers/guest-components/blob/main/image-rs/test_data/signature/cosign/cosign1.pub. 

This public key is corresponding to a private key that I generated test signed images myself and we use this for test for a long time in image-rs CI. We'd better to have a CI that generates signed image together with verification. See issue https://github.com/kata-containers/kata-containers/issues/9360

Reply:
> yes, this pubkey comes from https://github.com/confidential-containers/guest-components/blob/main/image-rs/test_data/signature/cosign/cosign1.pub.
> 
> This public key is corresponding to a private key that I generated test signed images myself and we use this for test for a long time in image-rs CI. We'd better to have a CI that generates signed image together with verification. See issue #9360

Ok. Until the CI fully handles the signing, I'd prefer the key to be downloaded from the guest-components repo, along with the explanation you provided here.

Reply:
I think this comment was not addressed ?
Not a big deal, but I'm pretty sure that in a couple months, nobody will remember where this key comes from :) Having it downloaded makes it clear what its origin is, and prevents problems if the key and test images are changed on the other side.

Reply:
An attempt was made - lets see if it works!